### PR TITLE
convert button toggle to use hooks

### PR DESCRIPTION
**Pull request by illiteratewriter**
- [X] Chore 
- [ ] Breaking change 
- [ ] There is an open issue which this change addresses
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [x] My commits follow the [Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My change requires a change to [Typescript typings](./types/lib).
  - [ ] I have updated the typings accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/__tests__/ButtonToggle.spec.js [hunk 1]
```diff
@@ -10,12 +10,6 @@ describe("ButtonToggle", () => {
     expect(wrapper.first(Button).shallow().text()).toBe("Ello world");
   });
 
-  it('should have button already toggled for defaultValue true', () => {
```
Comment (davidacevedo) [inline src/__tests__/ButtonToggle.spec.js:13]: Why was this removed?

File: src/ButtonToggle.js [hunk 2]
```diff
@@ -15,72 +15,52 @@ const defaultProps = {
   defaultValue: false,
```
Comment (davidacevedo) [inline src/ButtonToggle.js:15]: To avoid checking for `onClick`, `onBlur`, and `onFocus` you can add defaultProps to the component:
```
const noop = () => {}
const defaultProps = {
   ...,
   onBlur: noop,
   onFocus: noop,
   ...
}

Comment (davidacevedo) [reply src/ButtonToggle.js:15]: Although for 1:1 translation this works as the old implementation did this.

File: src/__tests__/ButtonToggle.spec.js (hunk 1 above)
Comment (illiteratewriter) [reply src/__tests__/ButtonToggle.spec.js:13]: This was removed since the state can be accessed only inside class components. Should we add `react-testing-library` to solve this, or are there any other suggestions?

Comment (davidacevedo) [reply src/__tests__/ButtonToggle.spec.js:13]: You can do this with enzyme by:
```
wrapper.find(Button).props().active
```
and run expectations on that.

Although I do recommend going with react-testing-library instead as enzyme is technically a dead project now.

**Review by davidacevedo — APPROVED** [review]
